packages/mdc-auto-init/index.ts

```diff
@@ -40,36 +54,39 @@ function _emit(evtType, evtData, shouldBubble = false) {
   document.dispatchEvent(evt);
 }
 
+/* istanbul ignore next: optional argument is not a branch statement */
 /**
- * Auto-initializes all mdc components on a page.
+ * Auto-initializes all MDC components on a page.
  */
-export default function mdcAutoInit(root = document, warn = CONSOLE_WARN) {
+export function mdcAutoInit(root = document, warn = CONSOLE_WARN) {
   const components = [];
-  const nodes = root.querySelectorAll('[data-mdc-auto-init]');
-  for (let i = 0, node; (node = nodes[i]); i++) {
-    const ctorName = node.dataset.mdcAutoInit;
+  const nodes: Element[] = [].slice.call(root.querySelectorAll('[data-mdc-auto-init]'));
+
+  for (const node of nodes) {
+    const ctorName = node.getAttribute('data-mdc-auto-init');
     if (!ctorName) {
       throw new Error('(mdc-auto-init) Constructor name must be given.');
     }
 
-    const Ctor = registry[ctorName];
-    if (typeof Ctor !== 'function') {
+    const Constructor = registry[ctorName]; // tslint:disable-line:variable-name
+    if (typeof Constructor !== 'function') {
       throw new Error(
         `(mdc-auto-init) Could not find constructor in registry for ${ctorName}`);
     }
 
-    if (node[ctorName]) {
+    if (Object.getOwnPropertyDescriptor(node, ctorName)) {
       warn(`(mdc-auto-init) Component already initialized for ${node}. Skipping...`);
       continue;
     }
 
     // TODO: Should we make an eslint rule for an attachTo() static method?
-    const component = Ctor.attachTo(node);
+    // See https://github.com/Microsoft/TypeScript/issues/14600 for discussion of static interface support in TS
```

Comment:
Is the TODO above this obsolete? (Like, would our build actually fail without an `attachTo` given the typings above? Plus we're using tslint now to reuse the Google configuration.)

Reply:
Unfortunately TypeScript won't catch that because it doesn't do static type checking.
See https://github.com/Microsoft/TypeScript/issues/14600.

`attachTo()` in the `Component` interface is only there to make TS happy when we try to call `Constructor.attachTo()` in `mdcAutoInit()`. It doesn't verify that `static attachTo()` exists on classes passed to `register()`.

It _does_, however, verify that the required constructor signature exists (via `new<F...>()`) on classes passed to `register()`.